framework/src/wifi_manager/wifi_manager.c

```diff
@@ -407,47 +444,66 @@ wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
 
 	if (g_dw_nwevent_mqfd == (mqd_t)ERROR) {
 		ndbg("iotivity connect event message queue init fail");
-		free(w_mutex);
-		w_mutex = NULL;
-		return WIFI_MANAGER_FAIL;
+		goto error_with_mutex_destory;
```

Comment:
I think you don't need to use 'error_with_mutex_destory'
anyway it's error, so developers are not using wifi-manager

Reply:
error_with_mutex_destory is required in case that releasing mutex is not needed.
That is, just destory without releasing.

Reply:
ok you're right